pkg/agent/config-updater.go

```diff
@@ -67,9 +67,11 @@ func (u *updater) Start() error {
 		for {
 			select {
 			case cfg := <-u.newConfig:
-				u.Info("scheduling update after %v rate-limiting timeout...", rateLimitTimeout)
-				pendingConfig = cfg
-				ratelimit = time.After(rateLimitTimeout)
+				if cfg != nil {
+					u.Info("scheduling update after %v rate-limiting timeout...", rateLimitTimeout)
+					pendingConfig = cfg
+					ratelimit = time.After(rateLimitTimeout)
+				}
```

Comment:
Should we log on an else branch the fact that we received an event (with explicitly nil config data, so the sender telling us that there is not config data) ? If that fact is already (somehow) logged on the sender side of the channel then logging here is not necessary... I just don't remember if it is.

Reply:
+1

Reply:
What about removing this change (so leaving the original comment in place) and just have the sending side (the other change in this pr) not send nil configuration (as we are ignoring it anyway)